Python/Product/Analysis/Interpreter/Ast/AstPythonInterpreterFactory.cs

```diff
@@ -441,30 +446,53 @@ bool useDefaultDatabase
 
         #region Module Imports
 
-        public enum TryImportModuleResult {
+        /// <summary>
+        /// Determines whether the specified directory is an importable package.
+        /// </summary>
+        public bool IsPackage(string directory) {
+            return ModulePath.PythonVersionRequiresInitPyFiles(Configuration.Version) ?
+                !string.IsNullOrEmpty(ModulePath.GetPackageInitPy(directory)) :
+                Directory.Exists(directory);
+        }
+
+        public enum TryImportModuleResultCode {
             Success,
             ModuleNotFound,
             NeedRetry,
             NotSupported,
             Timeout
         }
 
+        public struct TryImportModuleResult {
```

Comment:
Is it part of a protocol?

Reply:
You mean language server protocol? No. It's public API for anyone deriving their own interpreter factory (which is basically nobody but us).

Reply:
And it's intentionally scoped to `AstPythonInterpreterFactory`, not to interpreter factories in general. The protocol is `IPythonInterpreter.ImportModule`, which returns `null` if the module can't be imported (and we really need a new async version, but until it can be plumbed through everywhere there's basically no point)

Reply:
Ok, so it's a general purpose struct. Let's make it immutable then.